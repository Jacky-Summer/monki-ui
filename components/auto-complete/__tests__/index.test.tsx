import React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import AutoComplete, { AutoCompleteProps } from '../autoComplete'
import { waitForWrapperToPaint } from '../../../tests/utils'

const testArray = [
  { value: 'ab', number: 11 },
  { value: 'abc', number: 1 },
  { value: 'b', number: 4 },
  { value: 'c', number: 15 },
]

let wrapper: ReactWrapper
let inputNode: ReactWrapper

const testProps: AutoCompleteProps = {
  onSearch: query => testArray.filter(item => item.value.includes(query)),
  onSelect: jest.fn(),
  placeholder: 'This is a placeholder',
}

describe('AutoComplete', () => {
  beforeEach(() => {
    wrapper = mount(<AutoComplete {...testProps} />)
    inputNode = wrapper.find('input')
  })

  it('test basic AutoComplete behavior', () => {
    expect(inputNode.prop('placeholder')).toBe(testProps.placeholder)
    inputNode.simulate('change', { target: { value: 'a' } })
    waitForWrapperToPaint(wrapper, 1000)
    // should have two suggestion items
    expect(wrapper.find('.suggestion-item').length).toEqual(2)
    // click the first item
    wrapper
      .find('.suggestion-item')
      .first()
      .simulate('click')
    expect(testProps.onSelect).toHaveBeenCalledWith({ value: 'ab', number: 11 })
    expect(wrapper).toMatchSnapshot()
  })

  // TODO: test case - should provide keyboard support
  // TODO: test case - click outside should hide the dropdown
})
