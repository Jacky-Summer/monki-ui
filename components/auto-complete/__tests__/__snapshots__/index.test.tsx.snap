// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete test basic AutoComplete behavior 1`] = `
<AutoComplete
  onSearch={[Function]}
  onSelect={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "number": 11,
            "value": "ab",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  placeholder="This is a placeholder"
>
  <div
    className="mk-auto-complete"
  >
    <ForwardRef
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="This is a placeholder"
      value="ab"
    >
      <div
        className="mk-input-wrapper"
      >
        <input
          className="mk-input-inner"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="This is a placeholder"
          value="ab"
        />
      </div>
    </ForwardRef>
    <Transition
      animation="zoom-in-top"
      appear={true}
      in={false}
      timeout={300}
      unmountOnExit={true}
      wrapper={false}
    >
      <CSSTransition
        appear={true}
        classNames="zoom-in-top"
        in={false}
        timeout={300}
        unmountOnExit={true}
      >
        <Transition
          appear={true}
          enter={true}
          exit={true}
          in={false}
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={300}
          unmountOnExit={true}
        >
          <ul
            className="mk-suggestion-list"
          >
            <li
              className="suggestion-item"
              key="0"
              onClick={[Function]}
            >
              ab
            </li>
            <li
              className="suggestion-item"
              key="1"
              onClick={[Function]}
            >
              abc
            </li>
          </ul>
        </Transition>
      </CSSTransition>
    </Transition>
  </div>
</AutoComplete>
`;
